release-21.1: Fix some nilness linter findings

stevendanna opened the pull request:
Backport 2/6 commits from #61608.

I've backported just the fixes from 61608 rather than the linter addition
as well to avoid having to introduce dependency bumps.

/cc @cockroachdb/release

---

This fixes an assortment of minor issues found by the nilness linter.

Most appear to be err checks that were simply copy and pasted but 
were no longer needed.

Release note: None

stevendanna committed:
jobs,storage: don't call errors.Wrap-style functions with nil errors

These calls to errors.Wrap-style functions were passing errors that
would always be nil, which means that they would be returning a nil
error rather than the non-nil error that the author likely expected.

Release note: None

stevendanna committed:
multiple: fix uses of provably nil errors

These cases were found by the nilness linter. Most appear to be err
checks that were simply copy and pasted but were no longer needed.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62716)
<!-- Reviewable:end -->

knz commented:
Fwiw the dep bumps for linters are ok to backport generally.

knz committed:
server: simplify an error case

`NewAssertionErrorWithWrappedErrf` already includes the text of the
original error on the right of the wrapping message. No need to
include it a second time.

Release note: None

stevendanna commented:
@knz I'll take a look at backporting the linter in a follow up.  I've added your error simplification to this branch, let me know if you'd prefer it not be backported.

knz commented:
LGTM - I think this is ready to merge?

stevendanna commented:
Yup, was just held up by some previous CI flakes. Thanks for the reminder!
